### PR TITLE
Feature/e UI 3677 reassign page select phillip whitaker

### PR DESCRIPTION
**Pull request by phillip-whitaker**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-3677

### Change description ###

Added cancel link functionality and generic error

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts [hunk 1]
```diff
@@ -98,19 +102,28 @@ export class TaskAssignmentContainerComponent implements OnInit, OnDestroy {
   }
 
   public initForm(): void {
-    this.formGroup = new FormGroup({
-      caseWorkerName: new FormControl()
+    this.formGroup = this.fb.group({
+      caseWorkerName: ['', Validators.required]
```
Comment (paul-graham) [inline src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts:106]: `caseworkerName`? There's a fair bit of inconsistency around how `caseworker` appears within the app but it is definitively _one_ word, not two.

File: src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts [hunk 2]
```diff
@@ -15,46 +16,49 @@ import { getAssigneeName, handleFatalErrors } from '../../utils';
...
   public showManage: boolean = false;
-  public caseworker: Caseworker;
   public verb: TaskActionType;
 
   public successMessage: InfoMessage;
   public excludedCaseworkers: Caseworker[];
   public location: Location;
 
   public formGroup: FormGroup;
+  public caseWorkerErrorMessage: ErrorMessagesModel = null;
```
Comment (paul-graham) [inline src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts:37]: `caseworkerErrorMessage` - see other comment about `caseworkerName`.

File: src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts [hunk 3]
```diff
@@ -98,19 +102,28 @@ export class TaskAssignmentContainerComponent implements OnInit, OnDestroy {
...
+      }
       this.error = NAME_ERROR;
       return;
     }
-    this.error = null;
 
-    this.assignTask = this.taskService.assignTask(this.tasks[0].id, { userId: this.caseworker.idamId }).subscribe({
+    this.clearErrors();
+
+    const caseWorkerIdamId = this.formGroup.get('caseWorkerName').value;
```
Comment (paul-graham) [inline src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts:124]: `caseworkerIdamId` and `'caseworkerName'`.

**Review by paul-graham — CHANGES_REQUESTED** [review]
A couple of bits and pieces to tidy up.

File: src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts [hunk 4]
```diff
@@ -15,46 +16,49 @@ import { getAssigneeName, handleFatalErrors } from '../../utils';
 export const NAME_ERROR: ErrorMessage = {
   title: 'There is a problem',
   description: 'You must select a name',
```
[the remark this answers is not in the document]
Comment (phillip-whitaker) [reply src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts:18]: This is what is in the designs

File: src/work-allocation-2/containers/task-assignment/task-assignment-container.component.spec.ts [hunk 5]
```diff
@@ -95,32 +99,27 @@ describe('TaskAssignmentContainerComponent', () => {
...
-  it('should assign succesfully', () => {
-    const caseworker = mockCaseworkers[0];
-    component.caseworker = caseworker;
-    fixture.detectChanges();
+  it('should show banner error when continue clicked and form is invalid', () => {
+    component.formGroup.get('caseworkerName').setValue('');
+    component.formGroup.get('caseworkerName').updateValueAndValidity();
 
-    component.assign();
+    const continuteBtn = fixture.debugElement.nativeElement.querySelector('#continue__button');
```
Comment (paul-graham) [inline src/work-allocation-2/containers/task-assignment/task-assignment-container.component.spec.ts:116]: `continueBtn`?

File: src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (paul-graham) [reply src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts:18]: Fair enough. Still think it should be 'caseworker' and not 'name', though. 🤷‍♂️

File: src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts [hunk 6]
```diff
@@ -15,46 +16,49 @@ import { getAssigneeName, handleFatalErrors } from '../../utils';
...
-  fieldId: 'task_assignment_caseworker'
+  fieldId: 'case-worker-search__input'
```
Comment (paul-graham) [inline src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts:19]: There's a disparity between this and the id given to the input in the HTML, which is `'task_assignment_caseworker'`.

**Review by paul-graham — CHANGES_REQUESTED** [review]
Sorry, found a couple I missed the first time round...

**Review by paul-graham — APPROVED** [review]
Thanks for making the changes, @phillip-whitaker. 😁

**Pull request closed without merging**